src/gstring.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::cmp::Ordering;
+use std::ffi::{CStr, CString};
+use std::fmt;
+use std::ops::Deref;
+use std::os::raw::c_char;
+use std::ptr;
+use translate::*;
+use types::{StaticType, Type};
+use libc;
+
+use gobject_ffi;
+use ffi as glib_ffi;
+use value::{FromValueOptional, Value};
+
+#[derive(Hash)]
+pub struct GString {
+    s: Option<CString>,
+    ptr: *const c_char,
+    length: usize
+}
+
+impl GString {
+    pub unsafe fn new(ptr: *const c_char) -> Self {
+        assert!(!ptr.is_null());
+        let length = libc::strlen(ptr);
+        Self { s: None, ptr, length }
+    }
+
+    pub fn from(s: &[u8]) -> Self {
```

Comment:
`fn from` and `fn from_cstring` should go away and instead become `From<_>` impls. I think clippy also would warn about this

Reply:
You probably want `From` impls in both directions